channeldb/kvdb/backend.go

```diff
@@ -25,28 +32,93 @@ func GetBoltBackend(path, name string, noFreeListSync,
 	autoCompact bool) (Backend, error) {
 
 	dbFilePath := filepath.Join(path, name)
-	var (
-		db  Backend
-		err error
-	)
 
+	// Is this a new database?
 	if !fileExists(dbFilePath) {
 		if !fileExists(path) {
 			if err := os.MkdirAll(path, 0700); err != nil {
 				return nil, err
 			}
 		}
 
-		db, err = Create(BoltBackendName, dbFilePath, noFreeListSync)
-	} else {
-		db, err = Open(BoltBackendName, dbFilePath, noFreeListSync)
+		return Create(BoltBackendName, dbFilePath, noFreeListSync)
+	}
+
+	// This is an existing database. We might want to compact it on startup
+	// to free up some space.
+	if autoCompact {
```

Comment:
Hmm, don't think we need to do it at this very moment, but would be cool if we could have a "freshness" check. With the end goal of being able to detect possibly "empty" compaction iterations for nodes that restart frequently (dev nodes, etc).

Reply:
Good idea, I added the `--auto-compact-min-age` flag with a default value of 6 hours.